documentation/design-docs/data-catalog.md

```diff
@@ -0,0 +1,497 @@
+
+## Design
+
+After a decent amount of research and experimentation, we propose to build our
+data catalog and data integrity checks in **dbt**.
+
+dbt is a "Data Build Tool" that allows users to define and document ETL
+workflows using common software engineering patterns like version control
+and modularization. For a longer introduction, see [the
+docs](https://docs.getdbt.com/docs/introduction).
+
+For a fun take on what dbt is through the lens of what it is _not_, see
+[this blog post](https://stkbailey.substack.com/p/what-exactly-isnt-dbt).
+
+There are two distinct tools that are both somewhat confusingly referred to as
+"dbt":
+
+1. **dbt Cloud**: The SaaS version of dbt, including a web view with CI/CD, orchestration, monitoring, RBAC, and an IDE
+  * **Pros**: Provides a full-featured, hosted data catalog service
+  * **Cons**: Super expensive (minimum $100/seat/mo); owned and operated by a VC-funded startup
+2. **dbt Core**: Open source CLI that dbt Cloud is built on top of
+  * **Pros**: Free and open source; users have full control and ownership
+  * **Cons**: Basically only handles DAG builds (equivalent to e.g. CMake); orchestration/monitoring has to be handled by another tool
+
+dbt Cloud might save us time in the short run, but the process to get it
+approved and procured may take a number of months. It also seems risky to
+build our catalog on top of a VC-funded service, and the
+orchestration/monitoring tools it provides on top of the Core features
+are not high priority for us.
+As such, we think it would be  more prudent for us to build with dbt Core
+and design our own orchestration/monitoring/authentication integrations on top.
+Hence, when this doc refers to "dbt", we are actually referring to dbt Core.
+
+The downside of this choice is that we would have to choose a separate tool for
+orchestrating and monitoring our DAGs if we move forward with dbt. This is an
+important fact to note in our decision, because [orchestrators are notoriously
+controversial](https://stkbailey.substack.com/p/what-exactly-isnt-dbt):
+
+> Every other application, besides dbt, is an orchestrator. Every application
+> pays the price for it. Every person hates every application as soon as it
+> starts orchestrating. To orchestrate is to grapple with reality, to embrace
+> mortality, to welcome death.
+
+As such, we evaluate this choice with an eye towards options for third-party
+orchestration and monitoring.
+
+### Demo
+
+See the
+[`jeancochrane/dbt-spike`](https://github.com/ccao-data/data-architecture/tree/jeancochrane/dbt-spike/dbt)
+branch of this repo for an example of a simple dbt setup using our views.
+
+### Requirements matrix
+
+| Requirement | dbt | Notes |
+| ----------- | --- | ----- |
+| Referential integrity    | ✅ | Via tests. |
+| Schema consistency       | ✅ | Via tests. |
+| Documentation            | ✅ | Via docs generator. |
+| Orchestration/automation | ❌  | Requires dbt Cloud, or integration with another service. |
```

Comment:
Our orchestration needs are very simple (so far), so I think this makes sense. For now, we plan to use scheduled/triggered jobs in Glue + cron jobs where necessary.